tools/windows-node-installer/test/e2e/aws_test.go

```diff
@@ -1,90 +1,279 @@
 package e2e
 
 import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/ec2"
 	"github.com/openshift/windows-machine-config-operator/tools/windows-node-installer/pkg/cloudprovider"
 	awscp "github.com/openshift/windows-machine-config-operator/tools/windows-node-installer/pkg/cloudprovider/aws"
 	"github.com/openshift/windows-machine-config-operator/tools/windows-node-installer/pkg/resource"
 	"github.com/stretchr/testify/assert"
-	"os"
-	"testing"
+	"github.com/stretchr/testify/require"
 )
 
-// TestCreateDestroyWindowsInstance creates and terminates Windows instance on AWS.
-// TODO: After creation, check for the following properties on the instance:
-// - instance exist
-// - in running state
-// - has public subnet
-// - Public IP
-// - OpenShift cluster Worker SG
-// - Windows SG
-// - OpenShift cluster IAM
-// - has name
-// - has OpenShift cluster vpc
-// - instance and SG IDs are recorded to the json file
-// After termination, it checks for the following:
-// - instance terminated
-// - Windows SG deleted
-// - json file deleted
-func TestCreateDestroyWindowsInstance(t *testing.T) {
+var (
 	// Get kubeconfig, AWS credentials, and artifact dir from environment variable set by the OpenShift CI operator.
-	kubeconfig := os.Getenv("KUBECONFIG")
-	awscredentials := os.Getenv("AWS_SHARED_CREDENTIALS_FILE")
-	dir := os.Getenv("ARTIFACT_DIR")
+	kubeconfig     = os.Getenv("KUBECONFIG")
+	awscredentials = os.Getenv("AWS_SHARED_CREDENTIALS_FILE")
+	dir            = os.Getenv("ARTIFACT_DIR")
 
-	// The e2e test uses Microsoft Windows Server 2019 Base with Containers image, m4.large type, and libra ssh key.
 	// The CI-operator uses AWS region `us-east-1` which has the corresponding image ID: ami-0b8d82dea356226d3 for
 	// Microsoft Windows Server 2019 Base with Containers.
-	cloud, err := cloudprovider.CloudProviderFactory(kubeconfig, awscredentials, "default", dir,
-		"ami-0b8d82dea356226d3", "m4.large", "libra")
-	assert.NoError(t, err, "error creating clients")
+	imageID      = "ami-0b8d82dea356226d3"
+	instanceType = "m4.large"
+	sshKey       = "libra"
+
+	// awsProvider is setup as a variable for both creating, destroying,
+	// and tear down Windows instance in case test fails in the middle.
```

Comment:
What is the difference between destroying and tear down? I asked this question before and you did not answer.

Reply:
Oh ... I thought I am addressing in the comments saying "tear down Windows instance in case test fails in the middle" Which is what it is doing. Destroy is part of the test, tear down is if any required part of the test failed.